src/model.py

```diff
@@ -0,0 +1,33 @@
+import torch
+from diffusers import StableDiffusionPipeline
+
+
+class Model:
+    def __init__(self):
+        self.pipe = StableDiffusionPipeline.from_pretrained(
+            'CompVis/stable-diffusion-v1-4',
+            custom_pipeline='stable_diffusion_mega',
+        )
+        self.pipe.to('cuda' if torch.cuda.is_available() else 'cpu')
+        self.pipe.enable_attention_slicing()
+
+    def generate(self, pipeline, params):
```

Comment:
You can use `kwargs` from python to make it easier to use.
```python
def generate(self, pipeline: str, *, prompt: str = "", ...)
    ... 
```
And in app code:
```python
model.generate('text2img', **state)
```

I hope this will work with streamlit state. But at least it is better to decompose this function to parameter extraction and generation step.